metaflow/plugins/cards/card_cli.py

```diff
@@ -0,0 +1,388 @@
+from metaflow.client import Task
+from metaflow import JSONType, namespace
+from metaflow.exception import CommandException
+import webbrowser
+import re
+import click
+import os
+import signal
+import random
+from contextlib import contextmanager
+from functools import wraps
+from metaflow.exception import MetaflowNamespaceMismatch
+from .card_datastore import CardDatastore, NUM_SHORT_HASH_CHARS
+from .exception import (
+    CardClassFoundException,
+    IncorrectCardArgsException,
+    UnrenderableCardException,
+    CardNotPresentException,
+)
+
+from .card_resolver import resolve_paths_from_task, resumed_info
+
+
+def serialize_flowgraph(flowgraph):
+    graph_dict = {}
+    for node in flowgraph:
+        graph_dict[node.name] = {
+            "type": node.type,
+            "box_next": node.type not in ("linear", "join"),
+            "box_ends": node.matching_join,
+            "next": node.out_funcs,
+            "doc": node.doc,
+        }
+    return graph_dict
+
+
+def open_in_browser(card_path):
+    url = "file://" + os.path.abspath(card_path)
+    webbrowser.open(url)
+
+
+def resolve_card(ctx, pathspec, hash=None, type=None, follow_resumed=True):
+    """Resolves the card path based on the arguments provided. We allow identifier to be a pathspec or a id of card.
+
+    Args:
+        ctx : click context object
+        pathspec : pathspec
+        hash (optional): This is to specifically resolve the card via the hash. This is useful when there may be many card with same id or type for a pathspec.
+        type : type of card
+    Raises:
+        CardNotPresentException: No card could be found for the pathspec
+
+    Returns:
+        (card_paths, card_datastore, taskpathspec) : Tuple[List[str],CardDatastore,str]
+    """
+    if len(pathspec.split("/")) != 3:
+        raise CommandException(
+            msg="Expecting pathspec of form <runid>/<stepname>/<taskid>"
+        )
+
+    flow_name = ctx.obj.flow.name
+    run_id, step_name, task_id = None, None, None
+    # what should be the args we expose
+    run_id, step_name, task_id = pathspec.split("/")
+    pathspec = "/".join([flow_name, run_id, step_name, task_id])
+    try:
+        task = Task(pathspec)
+    except MetaflowNamespaceMismatch as e:
+        # todo : What to do in such cases where there is a namespace mismatch
+        namespace(None)
```

Comment:
why not just do namespace(None) initially? It seems like we do that anyways so no point doing it twice.

Reply:
Fixed this